views.py

```diff
@@ -0,0 +1,76 @@
+from django.shortcuts import render, render_to_response, get_object_or_404, redirect
+from django.http import HttpResponseRedirect, HttpResponse, JsonResponse
+from django.template.context_processors import csrf
+from django.template import RequestContext
+import json
+import requests
+import urllib.parse
+from collections import defaultdict
+import datetime
+import dateutil.parser
+import json
+
+def home(request):
+	if request.method == 'POST' and request.is_ajax():
+		username = request.POST['username']
+		year = datetime.datetime.now().year
+		# Starting date of the year
+		start_date = datetime.datetime(year, 1, 1, 00, 00, 00)
+		# Ending date of the year
+		end_date = datetime.datetime(year, 12, 31, 23, 59, 59)
+
+		generalParameters = {'action': 'query',
+		           'format': 'json',
+		           'list': 'usercontribs',
+		           'uclimit':'max',
+		           'ucuser':username,
+		           'ucdir':'newer',
+		           'ucnamespace':0,
+		           'ucstart':start_date.isoformat(' '),
+		           'ucend':end_date.isoformat(' ')
+		           }
+		articlesCreatedParameters = generalParameters.copy()
+		articlesEditedParameters = generalParameters.copy()
+		articlesCreatedParameters['ucshow'] = "new"
+		articlesEditedParameters['ucshow'] = "!new"
+		# For articles created
+		articlesCreatedResult, articlesCreatedMessage, articlesCreatedDates = getContributionTimelineData(articlesCreatedParameters)
+		articlesCreated = {'result':articlesCreatedResult, 'message':articlesCreatedMessage, 'dates':articlesCreatedDates}
+		# For articles edited
+		articlesEditedResult, articlesEditedMessage, articlesEditedDates = getContributionTimelineData(articlesEditedParameters)
+		articlesEdited = {'result':articlesEditedResult, 'message':articlesEditedMessage, 'dates':articlesEditedDates}
+		return HttpResponse(json.dumps({'articlesCreated':articlesCreated, 'articlesEdited':articlesEdited}), content_type="application/json")
+	return render(request, 'home.html', {})
+
+
+def getContributionTimelineData(parameters):
+	contributionDatesDict = defaultdict(int) # default value of int is 0
+	message = ""
+	result = True	# whenever result will be false, message will be shown to the user, telling what went wrong, instead of the graphs
+	url='https://en.wikipedia.org/w/api.php'
+	while True:
```

Comment:
While there is nothing wrong with this, you could make the code a little simpler by using exceptions:

    while True:
        r = requests.get(url, params=parameters)
        r.raise_for_status()
        json_data = json.loads(r.text)
        if 'error' in json_data:
            raise Error('Phew! Faced this error while creating graphs.\n Error: %s' + json_data['error']['info'])

and then wrap the calls to this method in a try/except block and return a HTTP 400 with the error message when an exception was raised. That way you can't have separate error handling for the 'created' and 'edited' charts, but that doesn't really seem useful anyway; OTOH your JS error handling logic will be simpler, with the error handling happening in the same place as the AJAX call.